### ISSUE TITLE
Objects do not round trip properly between YAML and YAML::XS [rt.cpan.org #41463]

### ISSUE THREAD

**Issue by ingydotnet** [ingydotnet/yaml-pm#92]
https://rt.cpan.org/Ticket/Display.html?id=41463

```
The Load() of both YAML and YAML::XS throw exceptions when confronted
with a dump produced by the other module. Please run the attached test
script to witness the problem first hand. 
```

**Comment by perlpunk** [comment]
Seems to me the only problem here is the zero-indented sequence:
```
use Math::BigFloat;
my $f = Math::BigFloat->new(3.14159265);
use YAML ();
use YAML::XS ();
say YAML::Dump($f);
say YAML::XS::Dump($f);

__END__
--- !!perl/hash:Math::BigFloat
_e:
  - 8
_es: '-'
_m:
  - 314159265
sign: +

--- !!perl/hash:Math::BigFloat
_e:
- 8
_es: '-'
_m:
- 314159265
sign: +
```

**Comment by perlpunk** [comment]
For convenience, here's the attached test script from RT:
```
#!/usr/bin/perl

use warnings;
use strict;

use Data::Dumper;
use Math::BigFloat;
use Test::More qw/no_plan/;
use YAML::XS ();
use YAML ();

my $f = Math::BigFloat->new (3.14159265);

my %loaders = (
    pp => sub { YAML::Load (@_) },
    xs => sub { YAML::XS::Load (@_) },
);

my %dumps = (
    pp => YAML::Dump ($f),
    xs => YAML::XS::Dump ($f),
);

warn Dumper \%dumps;

for my $dump (keys %dumps) {
    for my $loader (keys %loaders)  {
        my $f2;
        eval { $f2  = $loaders{$loader}->($dumps{$dump}) };
        diag "Exception during processing of a $dump dump with the $loader loader:\n\n$@" if $@;
        is_deeply ($f, $f2, "Round trip with a $dump dump and a $loader loader");
    }
}
```

**Comment by perlpunk** [comment]
Fixed by #207

**Comment by perlpunk** [comment]
Released YAML 1.25_001
